### PR TITLE
feat: modular branchName/prTitle/commitMessage templating

### PR DESCRIPTION
**Pull request by rarkins**
This PR refactors `branchName`, `commitMessage` and `prTitle` so that they are more easily editable and hopefully more understandable. By breaking each up into subsections, users can modify one part without needing to copy/paste the entire string.

Directly editing any of these fields will now be deprecated and a warning issued.

### CODE REVIEW

**Comment by rarkins** [comment]
`branchName`: `{{{branchPrefix}}}{{{managerBranchPrefix}}}{{{branchTopic}}}`

`managerBranchPrefix` defaults to empty string but is used selectively to distinguish between updates between different package managers.

`commitMessage`: `{{commitMessagePrefix}} {{commitMessageAction}} {{commitMessageTopic}} {{commitMessageExtra}} {{commitMessageSuffix}}`

`commitMessage` is broken into prefix - action - topic - extra - suffix

`prTitle` is no longer necessary to be distinct from `commitMessage`

**Comment by rarkins** [comment]
I would appreciate some extra eyes on this PR if anyone has the time. Cc @destroyerofbuilds @edmorley @adam-moss @hulkish and anyone else that currently overrides `commitMessage`, `branchName` or `prTitle`.

There is the risk of some perceived instability if people have "ignored" PRs (closed without merging) that now have a different branchName or prTitle, as those will be reopened. If the new title is "wrong" then it will require a patch to go away again, but if the new title is more "right" than the old one then it's something we'll keep (the PR is already open after all..).

I plan to merge this Wednesday morning Central European Time if there are no objections.

File: website/docs/_posts/2017-10-05-configuration-options.md [hunk 1]
```diff
@@ -178,15 +187,59 @@ For example, To add `[skip ci]` to every commit you could configure:
...
+This is used to alter `commitMessage` and `prTitle` without needing to copy/paste the whole string. Actions may be like 'Update', 'Pin', 'Roll back', 'Refresh', etc.
+
+## commitMessageExtra
+
+Extra description used after the commit message topic - typically the version.
+
+| name    | value  |
+| ------- | ------ |
+| type    | string |
+| default | Update |
```
Comment (destroyerofbuilds) [inline website/docs/_posts/2017-10-05-configuration-options.md]: Is the `default` actually `Update`?

File: website/docs/_posts/2018-04-09-configuration-templates.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+categories:
+  - deep-dives
+description: How to edit Renovate's config templates
+type: Document
+order: 30
+---
+
+Renovate has always been intended to be as configurable as possible and therefore supported handlebars templates for branch name, commit message, PR title and PR body from almost the very start. However, with great power also comes great potential to shoot yourself in the foot.
+
+Read on for some recommendations of what to do if you want to change an of the existing templates.
```
Comment (destroyerofbuilds) [inline website/docs/_posts/2018-04-09-configuration-templates.md]: `an` => `any`

Comment (destroyerofbuilds) [reply website/docs/_posts/2018-04-09-configuration-templates.md]: Though I understand, after reading both paragraphs, that the point of this article is to explain how to customize the branch name, commit message, etc., these paragraphs don't just come out and say `Here's how you customize the commit message, branch name, etc. for Renovate...`

File: website/docs/_posts/2018-04-09-configuration-templates.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+`commitMessagePrefix` is usually not necessary to configure directly, and is used by Renovate if it needs to add a prefix due to Semantic Commit conventions. Do not touch it unless you know what you're doing.
+
+`commitMessageAction` is usually just 1 word, e.g. 'Update', 'Pin', 'Refresh', etc. It's usually also not necessary to edit, although maybe you prefer 'Upgrade' instead of 'Update'?
+
+`commitMessageTopic` is usually 2-3 words aimed to identify _what_ is being updated. e.g. it might be `dependency react` or `Docker image ubuntu`. You may want to edit this, but if you think your idea/requirement is a good one then maybe you can propose it to the project or publish it as a preset config for others with similar requirements.
+
+`commitMessageExtra` usually refers to the version being updated to. e.g. `to v16` for a major upgrade, or `to v16.0.3` for a patch update. It may also be empty in some cases, e.g. if the action/topic is `Pin Docker digests`.
+
+`commitMessageSuffix` defaults to empty and is there for flexibility and future use. Maybe for `major` updates you always want the PR to end with `(MAJOR)`, for instance.
```
Comment (destroyerofbuilds) [inline website/docs/_posts/2018-04-09-configuration-templates.md]: What is the recommendation for adding a commit body? (After the commit summary, using newlines)

Comment (rarkins) [reply website/docs/_posts/2018-04-09-configuration-templates.md]: added

File: website/docs/_posts/2017-10-05-configuration-options.md (hunk 1 above)
Comment (rarkins) [reply website/docs/_posts/2017-10-05-configuration-options.md]: yes

Comment (destroyerofbuilds) [reply website/docs/_posts/2017-10-05-configuration-options.md]: I thought `commitMessageAction` was `Update`, while `commitMessageExtra` was `to 16.0.0`, or whatever?

Comment (rarkins) [reply website/docs/_posts/2017-10-05-configuration-options.md]: Yes, that's how it is in the final doc

**Pull request merged**